lib/spack/spack/installer.py

```diff
@@ -938,13 +957,13 @@ def __init__(
         # queue.
         if status == STATUS_REMOVED:
             raise InstallError(
-                f"Cannot create a build task for {self.pkg_id} with status '{status}'", pkg=pkg
+                f"Cannot create a task for {self.pkg_id} with status '{status}'", pkg=pkg
             )
-
         self.status = status
 
-        # Package is associated with a bootstrap compiler
-        self.compiler = compiler
+        # Getting the PID again because it will be needed for execute functionality.
+        # TODO: Should this be cached in PackageInstaller?
```

Comment:
Here's another `TODO` that needs to be re-evaluated.

Reply:
I don't think it's worth addressing -- any potential time saving by caching is is absolutely trivial compared to the install time. I'll remove the TODO.